roachtest: multitenant-upgrade failed

roachtest.multitenant-upgrade [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6549169?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/6549169?buildTab=artifacts#/multitenant-upgrade) on master @ [bbcb10f982e47943de63a3a2d1faf10d3eb49e75](https://github.com/cockroachdb/cockroach/commits/bbcb10f982e47943de63a3a2d1faf10d3eb49e75):


```
		Wraps: (2) cluster.RunE
		Wraps: (3) context canceled
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errors.errorString

	cluster.go:1933,multitenant_utils.go:144,panic.go:884,test_impl.go:259,cluster.go:1933,multitenant_utils.go:144,panic.go:884,test_impl.go:259,cluster.go:1933,multitenant_utils.go:144,panic.go:884,test_impl.go:265,sql_runner.go:176,sql_runner.go:205,multitenant_upgrade.go:314,multitenant_upgrade.go:162,multitenant_upgrade.go:35,test_runner.go:928: cluster.RunE: context canceled
		(1) attached stack trace
		  -- stack trace:
		  | main.(*clusterImpl).RunE
		  | 	main/pkg/cmd/roachtest/cluster.go:1950
		  | main.(*clusterImpl).Run
		  | 	main/pkg/cmd/roachtest/cluster.go:1931
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.(*tenantNode).stop
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/multitenant_utils.go:144
		  | runtime.gopanic
		  | 	GOROOT/src/runtime/panic.go:884
		  | main.(*testImpl).Fatal
		  | 	main/pkg/cmd/roachtest/test_impl.go:259
		  | main.(*clusterImpl).Run
		  | 	main/pkg/cmd/roachtest/cluster.go:1933
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.(*tenantNode).stop
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/multitenant_utils.go:144
		  | runtime.gopanic
		  | 	GOROOT/src/runtime/panic.go:884
		  | main.(*testImpl).Fatal
		  | 	main/pkg/cmd/roachtest/test_impl.go:259
		  | main.(*clusterImpl).Run
		  | 	main/pkg/cmd/roachtest/cluster.go:1933
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.(*tenantNode).stop
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/multitenant_utils.go:144
		  | runtime.gopanic
		  | 	GOROOT/src/runtime/panic.go:884
		  | main.(*testImpl).Fatalf
		  | 	main/pkg/cmd/roachtest/test_impl.go:265
		  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).Query
		  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:176
		  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).QueryStr
		  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:205
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.verifySQL
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/multitenant_upgrade.go:314
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.runMultiTenantUpgrade
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/multitenant_upgrade.go:162
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.registerMultiTenantUpgrade.func1
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/multitenant_upgrade.go:35
		  | main.(*testRunner).runTest.func2
		  | 	main/pkg/cmd/roachtest/test_runner.go:928
		  | runtime.goexit
		  | 	GOROOT/src/runtime/asm_amd64.s:1594
		Wraps: (2) cluster.RunE
		Wraps: (3) context canceled
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errors.errorString
```
<p>Parameters: <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_fs=ext4</code>
, <code>ROACHTEST_localSSD=true</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/multi-tenant
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*multitenant-upgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-19779

```
14:00:46 test_impl.go:323: test failure: 	sql_runner.go:176,sql_runner.go:205,multitenant_upgrade.go:314,multitenant_upgrade.go:162,multitenant_upgrade.go:35,test_runner.go:928: error executing 'SHOW CLUSTER SETTING version': pq: operation "show cluster setting version" timed out after 2m0s (given timeout 2m0s): value differs between local setting ([18 10 8 214 132 61 16 1 24 0 32 0]) and KV ([18 8 8 22 16 1 24 0 32 0]); try again later (<nil> after 1m59.305431352s)
```

--- a/pkg/cmd/roachtest/tests/BUILD.bazel
+++ b/pkg/cmd/roachtest/tests/BUILD.bazel
@@ -177,6 +177,7 @@ go_library(
         "//pkg/kv",
         "//pkg/roachpb",
         "//pkg/roachprod",
+        "//pkg/roachprod/config",
         "//pkg/roachprod/install",
         "//pkg/roachprod/logger",
         "//pkg/roachprod/prometheus",

--- a/pkg/cmd/roachtest/tests/multitenant_utils.go
+++ b/pkg/cmd/roachtest/tests/multitenant_utils.go
@@ -23,6 +23,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
 	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
 	"github.com/cockroachdb/cockroach/pkg/roachprod"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/config"
 	"github.com/cockroachdb/cockroach/pkg/security/username"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/stretchr/testify/require"
@@ -250,7 +251,7 @@ func startTenantServer(
 	errCh := make(chan error, 1)
 	go func() {
 		errCh <- c.RunE(tenantCtx, node,
-			append([]string{binary, "mt", "start-sql"}, args...)...,
+			append(append(append([]string{}, config.DefaultEnvVars()...), binary, "mt", "start-sql"), args...)...,
 		)
 		close(errCh)
 	}()

